src/usePresence.tsx

```diff
@@ -22,10 +22,13 @@ export function usePresence<T extends any>(
   }, [room, key]);
 
   //TODO: batch calls to this function before presence is ready
-  function set(value: T) {
-    if (!presence.current) return;
-    presence.current?.set(key, value);
-  }
+  const set = useCallback(
+    (value: T) => {
+      if (!presence.current) return;
+      presence.current?.set(key, value);
+    }, 
+    [presence],
```

Comment:
Does this work? My understanding here is because `presence.current` is what's changing, and not the js reference to `presence`, this won't rerender.

Reply:
So technically I think this "works" either way because we capture the original value of the ref, and it sees the update to inner pointer.

That said, I thought react might be smart about refs and re-create the function when `current` is updated though I didn't actually test that. I suppose it's more correct to pass `[]` as the deps?

Reply:
I think so, or at least, I think if having the presence there and not having it both do the same thing, I'd error on not having it?